crates/ruff_python_formatter/src/expression/string.rs

```diff
@@ -48,10 +48,19 @@ impl<'a> AnyString<'a> {
         match self {
             Self::Constant(_) => Quoting::CanChange,
             Self::FString(f_string) => {
+                let unprefixed = locator
+                    .slice(f_string.range)
+                    .trim_start_matches(|c| c != '"' && c != '\'');
+                let triple_quotes =
+                    unprefixed.starts_with(r#"""""#) || unprefixed.starts_with(r#"'''"#);
```

Comment:
If I understand this correctly, we need to figure out if this f-string is triple-quoted or not. If so, could we use `ruff_python_ast::str::is_triple_quote` here?

Reply:
That method only takes a prefix, so i'd need to extract the prefix here, which is surprisingly complex (iterators don't work because we don't want to allocate a new String when collecting)